### PR TITLE
Node playbooks improvements

### PR DESCRIPTION
**Pull request by Avi-Robusta**

### CODE REVIEW

File: playbooks/robusta_playbooks/k8s_resource_enrichments.py [hunk 1]
```diff
@@ -135,3 +136,44 @@ def get_resource_yaml(event: KubernetesResourceEvent):
...
+    resource = event.get_resource()
+    if not resource:
+        logging.error(f"status_enricher was called on event without a resource : {event}")
+        return
+    if not resource.status.conditions:
+        event.add_enrichment(
+            [MarkdownBlock(f"*No status/conditions to report for {resource.kind} {resource.metadata.name}*")]
+        )
+    headers = ["Type", "Status"]
+    if params.show_details:
```
Comment (arikalon1) [inline playbooks/robusta_playbooks/k8s_resource_enrichments.py:165]: Why do we need `show_details`?
Why not always show the message?

File: helm/robusta/values.yaml [hunk 2]
```diff
@@ -190,12 +192,25 @@ builtinPlaybooks:
   - job_events_enricher: {}
   - job_pod_enricher: {}
 
+- triggers:
+  - on_prometheus_alert:
+      alert_name: KubeletTooManyPods
+  actions:
+  - node_pods_capacity_enricher: {}
+  - alert_explanation_enricher:
+      alert_explanation: "You are approaching the limit of how many pods you can run on your node. This in and of its self is not an error but may impact the future scheduling of pods on this node."
```
Comment (arikalon1) [inline helm/robusta/values.yaml:201]: test suggestion:
explanation: "The node is approaching the maximum number of scheduled pods"
resolution: "Verify that you defined proper resource requests for your workloads. If pods cannot be scheduled, add more nodes to your cluster."

**Review by arikalon1 — APPROVED** [review]
nice work
left some comments

File: playbooks/robusta_playbooks/k8s_resource_enrichments.py [hunk 3]
```diff
@@ -135,3 +136,44 @@ def get_resource_yaml(event: KubernetesResourceEvent):
...
+def status_enricher(event: KubernetesResourceEvent, params: StatusEnricherParams):
+    """
+    Enrich the finding with the k8s objects's status conditions.
+
+    """
+    resource = event.get_resource()
+    if not resource:
+        logging.error(f"status_enricher was called on event without a resource : {event}")
+        return
+    if not resource.status.conditions:
```
[the remark this answers is not in the document]
Comment (Avi-Robusta) [reply playbooks/robusta_playbooks/k8s_resource_enrichments.py:160]: Sounds good

File: playbooks/robusta_playbooks/k8s_resource_enrichments.py (hunk 1 above)
Comment (Avi-Robusta) [reply playbooks/robusta_playbooks/k8s_resource_enrichments.py:165]: I saw we didn't always show the message for all of our enrichers (like for `node_status_enricher`) 
So I wanted to make it optional to be able to replace all of the status enrichers with this generic one.

File: playbooks/robusta_playbooks/node_enrichments.py [hunk 4]
```diff
@@ -28,6 +31,27 @@ def pod_row(pod: Pod) -> List[str]:
...
+        return
+
+    block_list: List[BaseBlock] = []
+    pod_list: PodList = Pod.listPodForAllNamespaces(field_selector=f"spec.nodeName={node.metadata.name}").obj
+
+    # the capacity limit is only relevant to currently running pods, not 'pending', 'succeeded' or 'failed'.
+    running_pod_count = len([pod for pod in pod_list.items if pod.status.phase.lower() == "running"])
+    pod_capacity = node.status.capacity.get("pods")
+    capacity_string = f" and the maximum capacity for pods on this node is {pod_capacity}" if pod_capacity else ""
+    pod_capacity_formatted_message = (
```
[the remark this answers is not in the document]
Comment (Avi-Robusta) [reply playbooks/robusta_playbooks/node_enrichments.py:48]: interesting, I like it, Ill merge it without it for now and try to add this + test tomorrow morning

File: helm/robusta/values.yaml (hunk 2 above)
Comment (Avi-Robusta) [reply helm/robusta/values.yaml:201]: added

File: playbooks/robusta_playbooks/k8s_resource_enrichments.py (hunk 1 above)
Comment (Avi-Robusta) [reply playbooks/robusta_playbooks/k8s_resource_enrichments.py:165]: For nodes its repetative
Screen Shot 2023-04-13 at 22 39 49

**Pull request merged**